pkg/proxy/benchmarking/bench_test.go

```diff
@@ -0,0 +1,36 @@
+package benchmarking
+
+import (
+	"fmt"
+	utilproxy "k8s.io/kubernetes/pkg/proxy/util"
+	"testing"
+)
+
+func BenchmarkWriter(b *testing.B) {
+	nRules := []int{10, 50, 100, 200, 500}
+
+	for _, n := range nRules {
+		b.Run(fmt.Sprintf("rule-%d", n), func(b *testing.B) {
+			for j := 0; j < b.N; j++ {
+				rules := utilproxy.LineBuffer{}
+				for i := 0; i < n; i++ {
+					//rules.Write(
+					//	"-A", "KUBE-SERVICES",
+					//	"-m", "comment", "--comment", `"test/default cluster IP"`,
+					//	"-m", "tcp", "-p", "tcp",
+					//	"-d", "10.96.11.12",
+					//	"--dport", "8000",
+					//	"-j", "KUBE-SVC-QWERTY")
+
+					rules.Write(
+						"-A KUBE-SERVICES",
+						"-m comment --comment", `"test/default cluster IP"`,
+						"-m tcp -p tcp",
+						"-d 10.96.11.12",
+						"--dport 8000",
+						"-j KUBE-SVC-QWERTY")
```

Comment:
We don't want to benchmark just a single random rule. We want to benchmark the rules _as the iptables proxier generates them_, so that we can test things like "if I make such-and-such change to how `externalTrafficPolicy: Local` services work, does that regress in performance?"

Even though the functions we're testing are in `utilproxy`, I think the benchmark will be in `pkg/proxy/iptables`, because that's mostly what we want to benchmark. We want (at least) a benchmark that is similar to `TestOverallIPTablesRulesWithMultipleServices`.